scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoInfer.scala

```diff
@@ -1,25 +1,39 @@
 package scalafix.internal.rule
 
+import metaconfig.{Conf, Configured}
+
 import scala.meta._
+import scalafix.internal.config.NoInferConfig
 import scalafix.lint.LintCategory
 import scalafix.lint.LintMessage
-import scalafix.rule.RuleCtx
-import scalafix.rule.SemanticRule
+import scalafix.rule.{Rule, RuleCtx, SemanticRule}
 import scalafix.util.SemanticdbIndex
 import scalafix.util.SymbolMatcher
 
-case class NoInfer(index: SemanticdbIndex)
+case class NoInfer(index: SemanticdbIndex,
+                   config: NoInferConfig = NoInferConfig.default)
     extends SemanticRule(index, "NoInfer")
     with Product {
-  private val badSymbol = SymbolMatcher.exact(NoInfer.badSymbols: _*)
+  private val badSymbol = SymbolMatcher.exact(NoInfer.badSymbols ++ config.extraSymbols: _*)
+
+  def this(index: SemanticdbIndex) =
+    this(index, NoInferConfig.default)
+
+  override def init(config: Conf): Configured[Rule] =
+    config
+      .getOrElse("noInfer")(NoInferConfig.default)
```

Comment:
I'm starting to think a good convention is to make the configuration key match exactly with the rule name (instead of lowercasing first character like I've done until now). What do you think?

Reply:
👍 to this

Reply:
I agree. It would feel more intuitive without the lower case.